src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -129,6 +300,17 @@ def apply(self, node1, node2, options=None):
         # Remove node2 from the parent
         node2.detach()
 
+        loop_var1 = node1.variable
+        loop_var2 = node2.variable
+
+        # If the loop variable isn't the same, we change all uses of the
+        # second loop's variable to reference the first loop's variable.
+        if loop_var1 != loop_var2:
+            references = node2.loop_body.walk(Reference)
+            for ref in references:
+                if ref.symbol == loop_var2:
+                    ref.symbol = loop_var1
+
```

Comment:
I was going to ask that we could do an assignment of "Assignment.create(Reference(loop_var2), Reference(loop_var1))", but this would defeat the purpose of what you want to do because then the additional assignment won't allow to collapse it with the parent loop. But maybe its still worth adding it a comment here and at the transformation docstring that this has the side effect that loop_var2 does not have the same value after the loop with the expectation that it isn't used.

Reply:
Added